test/mongodb.test.js

```diff
@@ -3200,6 +3200,38 @@ describe('mongodb connector', function() {
     });
   });
 
+  describe('Test encodeMongoDBURL function', function() {
+    const module = require('../');
+
+    context('mongodb protocol', function() {
+      it('when username:password is provided', function() {
+        const url = 'mongodb://@$/! +=*%:@$/! +=*%@host:999/db';
```

Comment:
This URL is ambiguous. It can be parsed as `{user: '@$/! +=*%', password: '@$/! +=*%'}` but also as `{user: '@$/! +=*%:@$/! +=*%', password: undefined'}` or even `{user: '', password: undefined, host: '@$/! +=*%:@$/! +=*%@host'}`.